core/src/main/java/bisq/core/xmr/connection/MoneroConnectionsManager.java

```diff
@@ -0,0 +1,169 @@
+import monero.common.MoneroRpcConnection;
+
+@Slf4j
+@Singleton
+public final class MoneroConnectionsManager {
+
+    private final Object lock = new Object();
+
+    private final MoneroConnectionManager connectionManager;
+    private final XmrConnectionList connectionList;
+
+    @Inject
+    public MoneroConnectionsManager(MoneroConnectionManager connectionManager,
+                                    XmrConnectionList connectionList) {
+        this.connectionManager = connectionManager;
+        this.connectionList = connectionList;
+        initialize();
+    }
+
+    private void initialize() {
+        synchronized (lock) {
+            loadConnections();
+            addDefaultConnection();
+        }
+    }
+
+    private void loadConnections() {
+        connectionList.getConnections().forEach(uriConnection ->
+                connectionManager.addConnection(toMoneroRpcConnection(uriConnection)));
+    }
+
+    private void addDefaultConnection() {
+        String defaultUri = WalletConfig.MONERO_DAEMON_URI;
+        if (!connectionList.hasConnection(defaultUri)) {
+            addConnection(UriConnection.builder()
+                    .uri(defaultUri)
+                    .username(WalletConfig.MONERO_DAEMON_USERNAME)
+                    .password(WalletConfig.MONERO_DAEMON_PASSWORD)
+                    .build());
+        }
+    }
+
+    public void addConnection(UriConnection connection) {
+        synchronized (lock) {
+            connectionList.addConnection(connection);
+            connectionManager.addConnection(toMoneroRpcConnection(connection));
+        }
+    }
+
+    public void removeConnection(String uri) {
+        synchronized (lock) {
+            connectionList.removeConnection(uri);
+            connectionManager.removeConnection(uri);
+        }
+    }
+
+    public UriConnection getConnection() {
+        synchronized (lock) {
+            return toUriConnection(connectionManager.getConnection());
+        }
+    }
+
+    public List<UriConnection> getConnections() {
+        synchronized (lock) {
+            return connectionManager.getConnections().stream().map(this::toUriConnection).collect(Collectors.toList());
+        }
+    }
+
+    public void setConnection(String connectionUri) {
+        synchronized (lock) {
+            connectionManager.setConnection(connectionUri);
+        }
+    }
+
+    public void setConnection(UriConnection connection) {
+        synchronized (lock) {
+            connectionManager.setConnection(toMoneroRpcConnection(connection));
+        }
+    }
+
+    public UriConnection checkConnection() {
+        synchronized (lock) {
+            // TODO: if autoSwitch is enabled, and the current connection reports as disconnected with this call,
+            //  the following getConnection() will return the status of the new connection after the switch
+            //  Is this the wanted behaviour, or should the old, now disconnected connection be returned?
```

Comment:
Yes, `checkConnection()` switches to the next best available connection if the current connection is disconnected and `autoSwitch` is enabled. The current connection after this call should be returned.